components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java

```diff
@@ -666,6 +670,99 @@ protected void unsetOrganizationUserResidentResolverService(
         OAuth2ServiceComponentHolder.setOrganizationUserResidentResolverService(null);
     }
 
+    @Reference(
+            name = "refreshtoken.grant.processor",
+            service = RefreshTokenGrantProcessor.class,
+            cardinality = ReferenceCardinality.OPTIONAL,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetRefreshTokenGrantProcessor"
+    )
+    protected void setRefreshTokenGrantProcessor(RefreshTokenGrantProcessor refreshTokenGrantProcessor) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Setting refresh token grant processor.");
+        }
+        OAuth2ServiceComponentHolder.getInstance().setRefreshTokenGrantProcessor(refreshTokenGrantProcessor);
+    }
+
```

Comment:
Let's add proper javadoc comments here. Do the same for other functions as well.

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2091/commits/d3dfa2211975383ff488e13e4313809db84573cf